### PR TITLE
Fix bug preventing Helm CRDs from being installed

### LINKED ISSUE #2497: ArgoCD failing to install CRDs in master from Helm Charts
Checklist:

* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

CRDs that are installed using the `crd-install` helm hook in master as of https://github.com/argoproj/argo-cd/commit/8e60bdfe598d9747383098e2fb62f90b0225dfdd don't seem to work.

I've looked at the manifest that was generated/cached in Redis and it does include the expected resources, so `helm template` is doing the right thing, however argo is not installing them; they're not even listed under resources to sync.

I've tried syncing using automated sync, manual sync in the UI, and manual sync in the CLI.

**To Reproduce**

Set up a cluster (Kind, k3s, etc..) with an ArgoCD install based on master, then install this application

https://gist.github.factset.com/dherman/5efe28ee3ca193af0649c40018cc9c3b

**Expected behavior**

CRDs should have been installed as part of a sync, and definitely should be listed under resources to sync.

**Screenshots**

**Version**

```shell
Paste the output from `argocd version` here.
```

argocd: v1.3.0+94112c9.dirty
  BuildDate: 2019-10-08T17:17:16Z
  GitCommit: 94112c970b0c8635f53145693088e9f428ab54c0
  GitTreeState: dirty
  GoVersion: go1.12.9
  Compiler: gc
  Platform: darwin/amd64
argocd-server: v1.3.0+88e9cf0
  BuildDate: 2019-10-11T15:49:08Z
  GitCommit: 88e9cf0e1813a0b328e5751bfb4de5df5dce9c40
  GitTreeState: clean
  GoVersion: go1.12.6
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: v0.13.1
  Kustomize Version: v3.1.0
  Helm Version: v2.12.1
  Kubectl Version: v1.14.0

**Logs**

```
Paste any relevant application logs here.
```

### PR DESCRIPTION
**Pull request by alexec**
Checklist:

* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)). 
* [x] Optional. My organisation is added to the README.

Closes #2497

### CODE REVIEW

File: test/e2e/helm_test.go [hunk 1]
```diff
@@ -64,20 +64,6 @@ func TestHelmHookDeletePolicy(t *testing.T) {
 		Expect(NotPod(func(p v1.Pod) bool { return p.Name == "hook" }))
 }
 
-func TestHelmCrdInstallIsCreated(t *testing.T) {
```
Comment (alexec) [inline test/e2e/helm_test.go]: this test was garbage

File: test/e2e/testdata/helm-crd/templates/crd.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+apiVersion: apiextensions.k8s.io/v1beta1
```
Comment (alexec) [inline test/e2e/testdata/helm-crd/templates/crd.yaml]: this is from Helm's examples

File: controller/sync.go [hunk 3]
```diff
@@ -578,13 +578,13 @@ func (sc *syncContext) pruneObject(liveObj *unstructured.Unstructured, prune, dr
 }
 
 func (sc *syncContext) hasCRDOfGroupKind(group string, kind string) bool {
-	for _, res := range sc.compareResult.managedResources {
-		if res.Target != nil && kube.IsCRD(res.Target) {
-			crdGroup, ok, err := unstructured.NestedString(res.Target.Object, "spec", "group")
+	for _, obj := range sc.compareResult.objs() {
```
Comment (alexmt) [inline controller/sync.go]: Looks like now method returns true if either live or target object is CRD. This is incorrect if target is nil and live is not nil. In this case crd won't be created so result should be false.

**Review by alexmt — CHANGES_REQUESTED** [review]

File: controller/sync.go (hunk 3 above)
Comment (alexec) [reply controller/sync.go]: I'll revert this.

**Comment by dcherman** [comment]
Yea sorry about that, I'm still waiting to hear back if I have approval to sign the CLA.  I'll see if I can push on that since it's holding up my other contribution :(

Thanks for addressing this bug!

**Comment by dcherman** [comment]
Just got approval for the CLA actually, so it's now signed.

**Review by alexmt — APPROVED** [review]

**Pull request merged**